File: MAP(Advanced Programming Methods)/LabAssig7/LabAssig7/Model/Statements/AssignStatement.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using LabAssig7.Model.Expressions;
using LabAssig7.Model.ADT;
namespace LabAssig7.Model.Statements
{
    public class AssignStatement:Statement
    {
        private string varName;
        private Expression expr;

        public AssignStatement(string var, Expression ex)
        {
            this.varName = var;
            this.expr = ex;
        }

        public PrgState Execute(PrgState p)
        {
            int res = this.expr.Eval(p.Dict);
            if(p.Dict.ContainsValue(this.varName))
            {
                p.Dict.UpdateValue(this.varName, res);
            }
            else
            {
                p.Dict.AddValue(this.varName, res);
            }
            return p;
        }

        public override string ToString()
        {
            return this.varName + "=" + this.expr;
        }
    }
}

File: MAP(Advanced Programming Methods)/lab2v2/src/Model/AssignStatement.java
package Model;

public class AssignStatement implements Statement {

    private String varName;
    private Expression expr;

    public AssignStatement(String varName,Expression expr)
    {
        this.varName=varName;
        this.expr= expr;
    }

    public PrgState execute(PrgState p)
    {
         IDictionary<String,Integer> dict = p.getSymbolTable();
         int result= this.expr.eval(dict);
         if(dict.contains(varName))
             dict.update(varName,result);
         else
            dict.add(varName,result);
         return p;

    }
    @Override
    public String toString()
    {
        return this.varName+"="+this.expr;
    }
}

